skglm/estimators.py

```diff
@@ -1,31 +1,201 @@
+    coefs, p_obj, kkt = solver(
+        X_, y, datafit, penalty, w, Xw, max_iter=model.max_iter,
+        max_epochs=model.max_epochs, p0=model.p0,
+        tol=model.tol,  # ws_strategy=model.ws_strategy,
+        verbose=model.verbose)
+
+    model.coef_, model.stop_crit_ = coefs, kkt
+    model.n_iter_ = len(p_obj)
+    model.intercept_ = 0.
+
+    return model
+
+
+def _glm_classif_fit(X, y, model, datafit, penalty):
+    check_classification_targets(y)
+    enc = LabelEncoder()
+    y = enc.fit_transform(y)
+    model.classes_ = enc.classes_
+    n_classes_ = len(model.classes_)
+    is_sparse = issparse(X)
+    if n_classes_ <= 2:
+        y = 2 * y - 1
+    X = check_array(X, accept_sparse="csc", dtype=np.float64, accept_large_sparse=False)
+    y = check_array(y, ensure_2d=False, dtype=X.dtype.type, accept_large_sparse=False)
+    check_consistent_length(X, y)
+
+    if y.ndim == 2 and y.shape[1] == 1:
+        warnings.warn("DataConversionWarning('A column-vector y"
+                      " was passed when a 1d array was expected")
+        y = y[:, 0]
+
+    if not hasattr(model, "n_features_in_"):
+        model.n_features_in_ = X.shape[1]
+
+    n_samples = X.shape[0]
+    if n_samples != y.shape[0]:
+        raise ValueError("X and y have inconsistent dimensions (%d != %d)"
+                         % (n_samples, y.shape[0]))
+
+    if not model.warm_start or not hasattr(model, "coef_"):
+        model.coef_ = None
+    if n_classes_ <= 2:
+        if isinstance(datafit, QuadraticSVC):
+            if is_sparse:
+                yXT = (X.T).multiply(y)
+                yXT = yXT.tocsc()
+            else:
+                yXT = (X * y[:, None]).T
+            X_ = yXT
+        else:
+            X_ = X
+
+        penalty = compiled_clone(penalty)
+        datafit_jit = compiled_clone(datafit, to_float32=X.dtype == np.float32)
+        if issparse(X):
+            datafit_jit.initialize_sparse(X_.data, X_.indptr, X_.indices, y)
+        else:
+            datafit_jit.initialize(X_, y)
+
+        if model.warm_start and hasattr(model, 'coef_') and model.coef_ is not None:
+            if isinstance(datafit, QuadraticSVC):
+                w = model.dual_coef_[0, :].copy()
+            else:
+                w = model.coef_[0, :].copy()
+            Xw = X_ @ w
+        else:
+            w = np.zeros(X_.shape[1], dtype=X_.dtype)
+            Xw = np.zeros(X_.shape[0], dtype=X_.dtype)
+
+        # check consistency of weights for WeightedL1
+        if isinstance(penalty, WeightedL1):
+            if len(penalty.weights) != X.shape[1]:
+                raise ValueError(
+                    "The size of the WeightedL1 penalty should be n_features, \
+                    expected %i, got %i" % (X_.shape[1], len(penalty.weights)))
+
+        coefs, p_obj, kkt = cd_solver(
+            X_, y, datafit_jit, penalty, w, Xw, max_iter=model.max_iter,
+            max_epochs=model.max_epochs, p0=X_.shape[1],
```

Comment:
@Klopfe why is p0 equal to X_.shape here ? Avoid writing such code without a TODO comment, it can very well go unnoticed, then be merged and it would take a long time to find it back

Reply:
I don't think that the "why" question is necessary. I did not leave a mistake on purpose. I was debugging the SVC. Now, you're right I should have been more cautious. I will pay attention. next time.

Reply:
I was asking if if was for debugging purposes or if there was a reason it was still there. Sorry if the tone was not friendly, it was not my intention.

Reply:
I misunderstood then. It should have been back to p0=10 and it could have been very long to find out. You're right

Reply:
I went back to `p0=model.p0,` should we put a different value for  the SVC?